analyzer/src/main/kotlin/managers/Yarn2.kt

```diff
@@ -0,0 +1,141 @@
+                        log.warn {
+                            "The Yarn2+ program '${yarnExecutable.name}' should be executable. Changing its rights."
+                        }
+                        require(yarnExecutable.setExecutable(true)) {
+                            "Cannot set the Yarn2+ program to be executable."
+                        }
+                    }
+                    yarnExecutable.absolutePath
+                } ?: throw IllegalStateException("No Yarn2+ executable could be found in 'yarnrc.yml'.")
+            }
+        } ?: "yarn2+ executable' in .yarn/releases/ and configured in .yarnrc.yml"
+    }
+
+    override fun getVersion(workingDir: File?): String =
+        // `getVersion` with a `null` parameter is called once by the Analyzer to get the version of the global tool.
+        // For Yarn2+, the version is specific to each definition file being scanned therefore a global version doesn't
+        // apply.
+        // TODO: An alternative would be to collate the versions of all tools in `yarn2CommandsByPath`.
+        if (workingDir == null) "N/A" else super.getVersion(workingDir)
+
+    override fun getVersionRequirement(): Requirement = Requirement.buildNPM(">=2.0.0")
+
+    override fun mapDefinitionFiles(definitionFiles: List<File>) = mapDefinitionFilesForYarn2(definitionFiles).toList()
+
+    override fun beforeResolution(definitionFiles: List<File>) =
+        // We depend on a version >= 2, so we check the version for safety.
+        definitionFiles.forEach { checkVersion(it.parentFile) }
+
+    override fun runInstall(workingDir: File): ProcessCapture =
+        // It should be enough for Yarn2+ to parse the lockfile to get the dependencies, and, if the former doesn't
+        // exist, run `yarn install --mode update-lockfile` to generate it (since yarn version 3.0.0-rc.10).
+        // Unfortunately, the lockfile doesn't contain the information required to build ORT's packages. An alternative
+        // could be to run `npm show --json` on each package present in the lockfile and rebuild a node_modules
+        // file hierarchy to be able to reuse the code present in the Npm package manager.
+        // Fortunately, an easier solution exist: by setting the option `nodeLinker` in the `yarnrc` file or as an
+        // environment variable, one can instruct Yarn to generate the node_modules files hierarchy.
+        run("install", workingDir = workingDir, environment = mapOf("YARN_NODE_LINKER" to "node-modules"))
+
+    override fun getRemotePackageDetails(workingDir: File, packageName: String): JsonNode {
+        // Yarn2+ 'info' command does not provide enough information so the default behavior of 'npm view --json' is
+        // kept. Note that it is mandatory to override this function as the super function calls `run()` which would
+        // call Yarn.
+        val process = delegateNpmAnalyzer.run(workingDir, "view", "--json", packageName)
```

Comment:
Wouldn't it make more sense to delegate to Yarn 1 and `yarn info --json` here?

Reply:
> Getting started with Yarn 2 requires you to have an up to date version of Node.js, at minimum Node.js 10.

From [1].
And when you have a Node installation, you have automatically NPM.
Therefore you always have NPM on an Yarn2 project. Right ?

[1]: https://snyk.io/blog/yarn-2-intro/